src/app-layer-parser.c

```diff
@@ -655,7 +662,7 @@ uint64_t AppLayerTransactionGetActiveLogOnly(Flow *f, uint8_t flags)
     AppLayerParserProtoCtx *p = &alp_ctx.ctxs[f->protomap][f->alproto];
 
     if (p->logger == TRUE) {
```

Comment:
The logger is for both directions, you can't control that. If you want it for one direction you have to ignore a certain direction in your log callback.  So nothing has really changed here.  On transaction complete, both log IDs will be incremented and we'll move forward.